File: algorithms/387. First Unique Character in a String/solution.py
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        letters = [chr(ord('a') + i) for i in xrange(26)]
        # Remember that collections.Counter is 100 times slower than s.count
        positions = [s.find(c) for c in letters if s.count(c) == 1]
        return min(positions) if positions else -1

# 104 / 104 test cases passed.
# Status: Accepted
# Runtime: 85 ms

File: algorithms/387. First Unique Character in a String/solution.cc
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> count(26);
        vector<int> last(26);
        for ( int i = 0; i < s.size(); ++i ) {
            ++count[s[i] - 'a'];
            last[s[i] - 'a'] = i;
        }
        int minIdx = s.size();
        for ( int i = 0; i < 26; ++i ) {
            if ( 1 == count[i] ) {
                minIdx = min(minIdx, last[i]);
            }
        }
        return minIdx == s.size() ? -1 : minIdx;
    }
};

// 104 / 104 test cases passed.
// Status: Accepted
// Runtime: 46 ms
